src/Parsers/ParserSelectQuery.cpp

```diff
@@ -264,6 +277,9 @@ bool ParserSelectQuery::parseImpl(Pos & pos, ASTPtr & node, Expected & expected)
             if (limit_with_ties_occured)
                 throw Exception("Can not use WITH TIES alongside LIMIT BY", ErrorCodes::LIMIT_BY_WITH_TIES_IS_NOT_SUPPORTED);
 
+            if (has_distinct_on)
```

Comment:
I suppose we can check if `limit_by_length` (or `limit_by_expression_list`) already initialized and do not need flag